docs/topics/control-flow.md

```diff
@@ -50,12 +50,58 @@ val max = if (a > b) {
 If you're using `if` as an expression, for example, for returning its value or
 assigning it to a variable, the `else` branch is mandatory.
 
-## When expression
+## When expressions and statements
 
-`when` defines a conditional expression with multiple branches. It is similar to the `switch` statement in C-like languages.
-Its simple form looks like this.
+`when` is a conditional expression that executes code based on multiple possible values or conditions. It is
+similar to the `switch` statement in C-like languages. For example:
 
 ```kotlin
+fun main() {
+    //sampleStart
+    val x = 2
+    when (x) {
+        1 -> print("x == 1")
+        2 -> print("x == 2")
+        else -> {
+            print("x is neither 1 nor 2")
+        }
+    }
+    //sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-conditions-when-statement"}
+
+`when` matches its argument against all branches sequentially until some branch condition is satisfied.
+
+You can use `when` in a few different ways. Firstly, you can use `when` either as an **expression** or as a **statement**.
+As an expression, `when` returns a value that you can use later in your code. As a statement, `when` completes an action
+without returning anything:
```

Comment:
_Mega-nitpick_: Maybe it is more about the developer not caring about the return value of that action? As of now, every Kotlin function returns something, but sometimes is the un-interesting `Unit` value.

Reply:
I updated the text to say "return anything of further use". WDYT?

Reply:
I like it, it's more exact without going into the nitty gritty details of statement vs. expression.